samtranslator/model/api/api_generator.py

```diff
@@ -237,6 +237,8 @@ def __init__(
         self.template_conditions = template_conditions
         self.mode = mode
 
+        self.swagger_editor = SwaggerEditor(self.definition_body) if self.definition_body else None
```

Comment:
A lot of the validation would happen in the `SwaggerEditor` constructor after these changes, so instantiating it in the `ApiGenerator` constructor will make the validation happen early. 
Also, this file used to instantiate multiple `SwaggerEditors` because it kept modifying `self.definition_body` and then would use the new value in each new instance of SwaggerEditor. Now, there's only one SwaggerEditor used and only at the end we use all of the changes made to update `self.definition_body` once.